### PR TITLE
Require compatible released integration versions

### PR DESCRIPTION
**Pull request by rogeriojorge**
Raise VMEX's sibling dependency floors to the releases that provide the APIs and fixes it uses. This remains a draft until the remaining minimum-version integration gates pass.

| Package | Required floor | Reason |
|---|---|---|
| booz_xform_jax | 0.4.0 | Correct registered-surface selection and cached-grid rebuilding; retains the magnetic-only JAX projection and asymmetric spectra. |
| SOLVAX | 0.21.0 | Accelerator-batched Thomas sweep; later releases add APIs VMEX does not yet require. |
| virtual-casing-jax | 0.0.7, unchanged | Align the compatibility comment and error hint with the existing requirement. |
| ESSOS | 0.17, unchanged | Existing field-line and coil APIs. |
| GKX | 1.8.0 | Adds the `tprim`/`fprim` drive fields used for explicit temperature/density gradients. |

Boozer 0.4.0 is released on PyPI. Its [release notes](https://github.com/uwplasma/booz_xform_jax/releases/tag/v0.4.0) distinguish the host-API fixes from the functional JAX API, which is unchanged from 0.3.0. No new universal transform-speedup claim is made.

### Completed checks

- The actual Boozer 0.4.0 wheel passed four upstream symmetric/asymmetric dense-reference tests, including magnetic-only values and bmnc JVP/VJP: [wheel evidence](https://github.com/uwplasma/vmex/pull/410#issuecomment-5755351706).
- Released Boozer 0.4.0 and SOLVAX 0.21.0, isolated together with Python 3.11.14/JAX 0.9.2 CPU/x64: all **21 Boozer-table, nine plotting, and eight omnigenity tests passed**, including full-marked tests and optimizer composition. Tests explicitly enabled JIT. [Detailed evidence and commands](https://github.com/uwplasma/vmex/pull/410#issuecomment-5755560071).
- Those numerical checks used source `f719c4ff`; the numerical source in this dependency-only branch is unchanged apart from the virtual-casing error hint.
- Updated packaging metadata: **11 passed**. Ruff and whitespace checks pass. The final Boozer-floor update removes five net lines from the dependency comment and adds no files.

The minimum-stack run found a real incompatibility at GKX 1.7.1: explicit
`r_over_lt`/`r_over_ln` failed because its `LinearParams` lacks `tprim` and
`fprim`. The tagged source and released 1.8.0 wheel provide these fields.
Commit `95710eb6` raises the turbulence-extra floor and updates installation
hints; it preserves the existing R/L to a/L conversion.

With GKX 1.8.0 and JAX/jaxlib 0.10.1, SciPy 1.16.0, SOLVAX 0.21.0,
Boozer 0.4.0, virtual-casing-jax 0.0.7, ESSOS 0.17 and NEO_JAX 1.0.2:

- `pip check` passes.
- Eleven packaging checks and four turbulence checks passed (15 total, no skips): explicit drive normalization, GKX geometry contract, growth-rate forward/reverse derivatives against finite differences, and eigenvector-proxy API guidance.
- Three real/synthetic surface API checks and three coil/exterior derivative checks passed on the earlier stack with GKX 1.7.1; these do not call GKX. [Commands and quadrature-warning limits](https://github.com/uwplasma/vmex/pull/410#issuecomment-5755788209).

The numerical source was main `45f3a7ae` plus this PR's changes. Ruff and whitespace checks pass. This is compatibility qualification, not a new performance or general turbulence-accuracy claim.

Additional minimum-stack checks passed with no skips: the NEO plotting diagnostic against its converged QI reference (1 test), and NESTOR full/skip consistency, a live tabulated-coil derivative, and boundary-Schur versus coupled-GCROT adjoints (3 tests, with `RUN_FULL=1`). These retain their case-specific accuracy scope; they do not establish a successful full finite-beta optimization.

### Remaining gates

The previous all-extras run used JAX 0.10.2, SOLVAX 0.21.0, Boozer 0.3.0, virtual-casing-jax 0.0.7, ESSOS 0.17, NEO_JAX 1.0.2 and GKX 1.7.1. It stopped after 201 passed and 11 skipped in the SOLVAX/free-boundary/ESSOS batch; that partial run is not a completed qualification. The newer Boozer checks do not replace it.

Finish the remaining free-boundary, virtual-casing and optional-dependency checks with the final exact floors. Required CI and review must pass. Commit `77a6b4b6` adds a bounded nightly minimum-version integration job using the exact released pins and real JIT-enabled surface, ESSOS, NEO and GKX calls. YAML and shell syntax pass; all seven tests in the combined selection passed with no skips (18.33 s). Latest-version CI still does not establish minimum-version compatibility. No release or merge is requested while these gates remain open.

Rebased final head: `c7ac449b` on main `45f3a7ae`. The tested combined checkout and this head have identical numerical source, metadata and tests; their only difference is the added nightly workflow. Static preflight passes: Ruff, mypy over 76 source files, documentation prose/media checks, and 86 guard tests. The three initially skipped built-HTML navigation checks were subsequently exercised: `sphinx-build -W -b html docs docs/_build/html` succeeds without warnings, and all six source/HTML navigation tests pass. No merge has been performed.

### CODE REVIEW

**Comment by rogeriojorge** [comment]
Handoff note on the booz_xform_jax floor.

booz_xform_jax 0.4.0 is merged (uwplasma/booz_xform_jax#15), tagged `v0.4.0` and has a GitHub release. At the time of writing its `Publish PyPI release` run was still queued and PyPI served 0.3.0. Confirm before touching the floor:

```bash
curl -s https://pypi.org/pypi/booz_xform_jax/json | python3 -c 'import sys,json;d=json.load(sys.stdin);print(d["info"]["version"])'
```

Decide the floor on what 0.4.0 actually contains. Its `jax_api.py` is byte-for-byte identical to 0.3.0, so the functional kernel vmex calls (including `magnetic_only`, which this PR's 0.3.0 floor is justified by) gains nothing. Everything new is in the `Booz_xform` class: `register_surfaces()` now selects surfaces, cached mode lists and grids are rebuilt when `mboz`/`nboz`/`nfp`/`asym` change between runs, and `__version__` is correct (0.1.1-0.3.0 report "0.1.0"). Raise the floor to 0.4.0 only if a vmex path uses the `Booz_xform` class and depends on those fixes; otherwise 0.3.0 is the honest floor. Either way, never compare against `booz_xform_jax.__version__` on 0.3.0 or older; read the installed distribution metadata.

**Comment by rogeriojorge** [comment]
The 0.4.0 PyPI wheel is now available. An isolated import of the released wheel passed the four cases of upstream `tests/test_jax_api.py::test_separable_projection_matches_dense_reference` from tag `v0.4.0`: symmetric and asymmetric fixtures, each with `magnetic_only=False/True` (4 passed in 14.68 s, CPU, Python 3.11.14, JAX 0.9.2, x64).

These tests compare output spectra and JVP/VJP with respect to the input `bmnc` spectrum against the independent dense projection at rtol=1e-10, atol=1e-13. They also check the reduced magnetic-only output keys. The imported package was verified to come from the extracted PyPI wheel, and the magnetic-only keyword is present. Both fixtures match the tagged Git blobs exactly.

Wheel SHA256: `845fc98616f8757d7cf4e1e4f13bca581bf8a150c10f3168518b1dbb3c24bc44`.

Reproduction: install `booz_xform_jax==0.4.0` into an isolated environment with the stated JAX version and x64 enabled, then run the named upstream test from tag `v0.4.0`, ensuring imports resolve to the wheel rather than an editable source tree.

This closes the wheel availability and bounded projection/AD check. It does not replace this PR's outstanding VMEX Boozer, plotting, omnigenity or exact-minimum-dependency integration suites; the runtime above is a test duration, not a benchmark speedup.

**Comment by rogeriojorge** [comment]
Further released-wheel integration qualification at source `f719c4ff` (the source and selected tests are byte-identical in plan commit `3a12fa01`):

- Isolated released wheels: **booz_xform_jax 0.4.0 + SOLVAX 0.21.0**, with Python 3.11.14, JAX/jaxlib 0.9.2, CPU/x64, one OpenMP/OpenBLAS thread. Imported module locations and installed versions were checked before testing; the existing environment was not modified.
- `python -m pytest -q -o addopts= tests/test_boozer_tables.py`: **21 passed in 47.24 s**, no skips. The module explicitly enables JIT. Coverage includes symmetric/asymmetric table construction, WOUT comparisons, field-strength preservation, straight-field-line conventions, and a live boundary JVP.
- Boozer file generation, surface selection and Boozer figure rendering also passed: **3 passed, 6 deselected in 3.34 s**, using `tests/test_plotting_boozer.py -k "run_booz or plot_boozmn"` with `-o usefixtures=_module_jit_enabled`. The remaining WOUT plotting checks then passed separately: **6 passed, 3 deselected in 14.60 s** with the complementary selection and same JIT setting. Together these cover all nine tests in that plotting module.
- Full-marked omnigenity spectrum/derivative checks were enabled with `RUN_FULL=1`: `tests/test_omnigenity.py -k "not least_squares_implicit_composes"` passed **7 tests, 1 deselected in 84.86 s**. The separately selected `test_least_squares_implicit_composes` then passed in **73.63 s**. Together these runs cover all eight tests in that module, with full-marked tests enabled.
- One expected warning states that the requested VMEC auxiliary text artifact is unsupported; the solve continues.

This extends the upstream dense-projection wheel check with VMEX integration. It does not certify the remaining free-boundary or all-extras minimum-version suite. Those gates remain open; no merge or floor-wide compatibility claim follows from these bounded integration tests alone.

**Comment by rogeriojorge** [comment]
Update: booz_xform_jax 0.4.0 is now live on PyPI (`curl` check above prints 0.4.0). The floor decision in the previous comment still applies: 0.4.0 only if a vmex path depends on the `Booz_xform` class fixes, otherwise keep 0.3.0.

**Comment by rogeriojorge** [comment]
### Surface API on the minimum-version stack

All three tests in `tests/test_virtual_casing_api.py` pass, with no skips (5.91 s). This includes a real coarse equilibrium followed by `surface_field_data_from_state`, the call repaired by #409. The elapsed time is test-run bookkeeping, not a speedup measurement.

Source: main `45f3a7aeaeedd50ee7a04d553bc0d281bb8510a4` with this PR's two commits (`80a029b2`, `d1f71ee4`) applied. Released dependencies: JAX/jaxlib 0.10.1, SciPy 1.16.0, SOLVAX 0.21.0, booz_xform_jax 0.4.0, virtual-casing-jax 0.0.7, ESSOS 0.17, NEO_JAX 1.0.2, GKX 1.7.1. `pip check` passes, and imports were verified against this combined source.

The all-extras stack uses JAX 0.10.1 because the turbulence extra requires it; the earlier Boozer-only checks used the base JAX 0.9.2 floor. These are different qualification configurations.

Run from the combined checkout after installing those exact versions:

```sh
JAX_PLATFORMS=cpu JAX_ENABLE_X64=1 OMP_NUM_THREADS=1 OPENBLAS_NUM_THREADS=1 \
python -m pytest -q -o addopts= -o usefixtures=_module_jit_enabled \
  tests/test_virtual_casing_api.py
```

The explicit fixture enables JIT, which ordinary unit-test setup disables. This is public-call integration evidence; it does not qualify coil derivatives, finite-beta optimization, GPU execution, or all optional dependencies. Those checks remain pending.

Three additional JIT-enabled checks passed with no skips (50.27 s):

```sh
JAX_PLATFORMS=cpu JAX_ENABLE_X64=1 OMP_NUM_THREADS=1 OPENBLAS_NUM_THREADS=1 \
python -m pytest -q -o addopts= \
  tests/test_virtual_casing_physics.py::test_synthetic_surface_gradient_fd_validates \
  tests/test_virtual_casing_physics.py::test_finite_beta_extender_field_and_gradient_outside_lcfs \
  tests/test_virtual_casing_physics.py::test_parameterized_extender_vjp_matches_rebuilt_surface_fd
```

These exercise ESSOS coil-direction AD versus finite differences, exterior spatial derivatives, and a moving-surface VJP against a rebuilt-surface difference. The exterior test emitted two accuracy warnings: estimated order-1 quadrature error reached 2.9e-3 against a requested 1e-3 on its coarse grid. Passing derivative consistency does not establish that tighter quadrature accuracy; the warnings remain part of this evidence. The module enables JIT explicitly.
